sql/wsrep_mysqld.cc

```diff
@@ -2019,7 +2019,26 @@ static int wsrep_TOI_begin(THD *thd, const char *db_, const char *table_,
   size_t buf_len(0);
   int buf_err;
 
+  DEBUG_SYNC(thd, "wsrep_TOI_begin_before");
+  mysql_mutex_lock(&thd->LOCK_thd_data);
+
+  /*
+    Setting wsrep_skip_wsrep_hton=true has a side-effect that this
+    query/thread cannot be killed.
+  */
   thd->wsrep_skip_wsrep_hton= true;
```

Comment:
If we are taking `thd->LOCK_wsrep_thd` for reading, then shouldn't we acquire `thd->LOCK_wsrep_thd` for writing it as well?

Reply:
Actually, should probably change the read to use LOCK_thd_data